### ISSUE TITLE
Support for static binary rewriting on ARMv8?

### ISSUE THREAD

**Issue by violet73** [dyninst/dyninst#1277]
I saw "Note that Dyninst currently does not support static binary rewriting on ARMv8." in [DyninstAPI ARMv8 status](https://github.com/dyninst/dyninst/wiki/DyninstAPI-ARMv8-status). Does the current version of Dyninst support static binary rewriting on ARMv8? 

I tried to instrument the call points of a function on ARMv8, but this led to incorrect program results.

[tracefunc.txt](https://github.com/dyninst/dyninst/files/8875393/tracefunc.txt)

[tracelib.txt](https://github.com/dyninst/dyninst/files/8875394/tracelib.txt)

[InterestingProgram.txt](https://github.com/dyninst/dyninst/files/8875396/InterestingProgram.txt)

./tracefunc/trace_func InterestingProgram InterestingProcedure

./InterestingProgram-rewritten

[res.txt](https://github.com/dyninst/dyninst/files/8875402/res.txt)

**Comment by hainest** [comment]
>Does the current version of Dyninst support static binary rewriting on ARMv8?

No. The person working on that left a few months ago. It's unlikely that will be implemented until we can find another person to work on the ARM port.

>I tried to instrument the call points of a function on ARMv8, but this led to incorrect program results.

What version of Dyninst are you using?
What hardware architecture are you building on?
What OS are you building on?
What CMake options did you use to build Dyninst?
Can you post the compiler build options you used to build 'InterestingProgram'?

**Comment by violet73** [comment]
> No. The person working on that left a few months ago. It's unlikely that will be implemented until we can find another person to work on the ARM port.

Got it. The problem I'm having is instrumenting function with static binary rewriting in ARMv8.

> What version of Dyninst are you using?

v12.1.0 and [master](https://github.com/dyninst/dyninst/commit/5e601431484bba3623697c37078ba794998c404c).

> What hardware architecture are you building on?

`lscpu` results：
```
Architecture:                    aarch64
CPU op-mode(s):                  64-bit
Byte Order:                      Little Endian
CPU(s):                          64
On-line CPU(s) list:             0-63
Thread(s) per core:              1
Core(s) per socket:              32
Socket(s):                       2
NUMA node(s):                    2
Vendor ID:                       Cavium
Model:                           2
Model name:                      ThunderX2 99xx
Stepping:                        0x1
CPU max MHz:                     2500.0000
CPU min MHz:                     1000.0000
BogoMIPS:                        400.00
L1d cache:                       2 MiB
L1i cache:                       2 MiB
L2 cache:                        16 MiB
L3 cache:                        64 MiB
NUMA node0 CPU(s):               0-31
NUMA node1 CPU(s):               32-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Vulnerable
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics cpuid asimdrdm
```

> What OS are you building on?

Linux thunderx2 5.4.0-99-generic #112-Ubuntu SMP Wed Feb 2 17:13:12 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux

> What CMake options did you use to build Dyninst?

I installed Dyninst directly using spack with no other customization options set.
- spack install dyninst@master
- spack install dyninst@12.1.0

> Can you post the compiler build options you used to build 'InterestingProgram'?

I used the [examples](https://github.com/dyninst/examples) compile script to build 'InterestingProgram'. The build options I found in  `CMakeCache.txt` were '-O3' and '-DNDEBUG'.

The GCC version is `gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)`.
